code/modules/worldgen/GenerateMining.dm

```diff
@@ -86,15 +86,16 @@ TYPEINFO(/turf/variableTurf/clear)
 		sound_environment = 22
 		ambient_light = TRENCH_LIGHT
 
-/proc/decideSolid(var/turf/current, var/turf/center, var/sizemod = 0)
-	if(!current || !center || (current.loc.type != /area/space && !istype(current.loc , /area/allowGenerate) && !isgenplanet(current)) || !istype(current, /turf/space))
-		return 0
-	if(ISDISTEDGE(current, AST_MAPBORDER))
-		return 0
-	var/probability = 100 - (((abs(center.x - current.x) + abs(center.y - current.y)) - (AST_MINSIZE+sizemod)) * AST_REDUCTION) + rand(-AST_TILERNG,AST_TILERNG)
-	if((abs(center.x - current.x) + abs(center.y - current.y)) <= (AST_MINSIZE+sizemod) || prob(probability))
-		return 1
-	return 0
+// this macro determines if a turf is to be considered for mining asteroid generation
+#define decideSolid(current, center, sizemod)\
```

Comment:
please provide proof via tracy that this significantly impacts startup and is worth it